src/Components/Components/src/ComponentFactory.cs

```diff
@@ -1,44 +1,37 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.AspNetCore.Components.Reflection;
 using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
+using Microsoft.AspNetCore.Components.Reflection;
 
 namespace Microsoft.AspNetCore.Components
 {
     internal class ComponentFactory
```

Comment:
@SteveSandersonMS \ @rynowak, for this release, I'm leaving these caches be static and avoiding designing components requiring services and possibly the shape of these types for 3.0.

Reply:
Sounds good to me.

Since it's not actually `static` (it's a singleton instance), there's a little bit of extra risk that with some future change, we might not remember it's a singleton and might add some code that's not thread-safe. Could we have a comment somewhere at the top warning that all code in here must be thread safe because there's just one globally-shared instance?